src/ReachSets/reach.jl

```diff
@@ -79,6 +81,7 @@ function reach(S::AbstractSystem,
         kwargs_dict[:template_directions_init])
     block_sizes = compute_block_sizes(partition)
 
+    # Check intersection with forbidden states
```

Comment:
?

Reply:
Seems that I didn`t move it from the very first version of implementation. I believe we should check intersection with forbidden states before pushing initial set for the next location into waiting list.

Reply:
This is a design decision. The algorithm should have one of the two following invariants:
1. *All* sets in the waiting list have already been checked for intersection with the bad states (before pushing them to the waiting list).
This version has the advantage that you can terminate earlier if you only look for *one* violation.
2. *No* set in the waiting list has been checked for intersection with the bad states (and so you have to check after popping a set from the waiting list).
This version has the advantage that you only need to check intersection at one place (less error-prone); you can alleviate this disadvantage of the first version by wrapping the `push!` to the waiting list in a separate function like `check_intersection_and_push!`.